doc/embedded/crosscompile_armv7.md

```diff
@@ -0,0 +1,151 @@
+algorithm from the command line. In today's tutorial, we are going to compile
+the k-NN app, so we can run a couple of them from the command line on a Raspbery
+PI 2.
+
+Since the PI 2 has 1 GB of RAM, depending on the OS you are using you might find
+at least 250 MB of these are already used by default. This leaves us with around
+750 MB of RAM, this is not enough to compile all the apps that we already have,
+and given the fact that we only dispose of one ARMv7 core clocked at 1 GHZ,
+it might take a lot of time for the compilation to finish. As a result it will be easier
+to cross-compile all mlpack apps on your hardware including the integration tests, and then
+move all them directly to the Raspberry PI.
+
+The first step is to download a cross compiler toolchain that is going to run on an
+x86\_64 machine referred to as the host and produce binaries that runs on an ARMv7
+referred to as the target. There are several toolchains around, but my
+favorites are the ones produced by [Bootlin](https://bootlin.com/), they have
+produced toolchains for all kind of architectures, ARMv7, ARMv8, Sparc, MIPS,
+RISC-V, etc, with three different sets of compilers. To get the toolchain please
+visit the following website: `https://toolchains.bootlin.com/`
+
+You need to select the architecture and the compiler set. In our case, we are
+looking for `armv7-eabihf`, and we will use `glibc`, you can also
+use `musl` or `uclibc` if you are interested in an optimized apps footprints.
+However, in this tutorials, we will stick to glibc for simplicity.
+
+If you want to get a link directly to the toolchain and download it with `wget`
+use the following command:
+
+```
+wget https://toolchains.bootlin.com/downloads/releases/toolchains/armv7-eabihf/tarballs/armv7-eabihf--glibc--stable-2024.02-1.tar.bz2
+```
+
+Once we have downloaded the toolchain we can extract the content using the
+following command:
+
+```
+tar -xvf armv7-eabihf--glibc--stable-2024.02-1.tar.bz2
+```
+
+Once we have done that, now the toolchain is ready, feel free to explore inside
+the toolchain you will find the basic set of compilers in addition to a sysroot
+folder that contains standard libraries headers.
+
+Now, we aim to use this toolchain to compile mlpack. The first thing 
+is to get the source code of the library, either you can download the zip
+file of the last version on github or basically clone the main branch. In this
+case I will clone the main branch:
+
+```
+git clone git@github.com:mlpack/mlpack.git
+```
+
+Once we have downloaded the source code, now we need to get mlpack external
+dependencies. mlpack has three dependcies, ensmallen which
+provides mathematical solvers and optimizers for training, Cereal for model
+serialization, and Armadillo for linear algebra and matrix manipulation.
+Armadillo itself can use several backends, in this tutorials we will use
+OpenBLAS since it is optimized on a broad range of different architectures.
```

Comment:
In the commit I linked to I added a little blurb indicating that everything is header-only *except* OpenBLAS which must be cross-compiled, and mentioned that the autodownloader does that step automatically.  I don't know if it's needed, but I suppose it could be nice to add a bit of documentation on how you might manually cross-compile OpenBLAS... but I don't think it's a huge deal either way.